### DISCUSSION TITLE
GitHub Pages building with Jekyll despite .nojekyll present

### DISCUSSION THREAD

**Discussion by egouldo** [community/community#73729] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I am building a GitHub page using this [workflow file](https://github.com/egouldo/ManyEcoEvo/actions/runs/6779032303/workflow).

I am building the site to the `docs/` directory using this workflow, and then wish to deploy this directory. My workflow file uses the action JamesIves/github-pages-deploy-action@v4.4.1 to deploy. This is successful (https://github.com/egouldo/ManyEcoEvo/actions/runs/6779032303), however, immediately afterwards, the GitHub pages bot attempts to rebuild my site using Jekyll, and then fails https://github.com/egouldo/ManyEcoEvo/actions/runs/6779117783. This is despite there being a .nojekyll file in the target directory, which is written there by the original workflow step, https://github.com/egouldo/ManyEcoEvo/blob/ffaa68e544eb9fc137948aa6f9bc8b47a7af0d9f/.github/workflows/pkgdown.yaml#L44-L46.

I have checked that the .nojekyll file is present on the gh-pages branch. What am I missing?

**Comment by b8zeek** [comment, marked as answer]
Hey. 👋🏼 

I took a look at the log from your build...

This error message "No such file or directory @ dir_chdir - /github/workspace/docs" indicates that the directory `/github/workspace/docs` doesn't exist in your repository and that is causing the build to fail. So, you should make sure the `docs` directory exists in your repository, otherwise create it. Pay attention that the name is correct because the system is case-sensitive.

Good luck. 🍺 

🍀

**Reply by egouldo** [reply]
Thank you!! That's incredibly helpful. I think I have misunderstood exactly what is happening when the GitHub-bot build kicks in. I had my GitHub Pages settings to deploy from `gh-pages/docs/` but I needed it set to `/root/`. Whilst my workflow creates a `/docs/` folder in the main branch, only the contents of `/docs/` are copied into the root of the `gh-pages` branch! I've just forced a rebuild and the page has successfully deployed.

**Reply by b8zeek** [reply]
Amazing Elliot! 🚀 

I am glad I was able to help you; love to see this! ☺️

![Screenshot 2023-11-07 at 10 29 54](https://github.com/community/community/assets/24759563/08cc2eec-cac9-4c74-a6d4-83bed230c221)

Take care bro. 🍺 

🍀

**Reply by queenofcorgis** [reply]
Nice work @bejzik8 🤩 - thanks for the assist and thanks for marking the answer @egouldo ❤️

**Reply by b8zeek** [reply]
I am glad I was able to assist! I am learning through helping; and I'm enjoying it. 😊

Thanks for the support stranger! 🙏

🍀
